### ISSUE TITLE
[Proposal] Implement OpenAI's TopK Auxilliary for preventing dead latents. 

### ISSUE THREAD

**Issue by jbloomAus** [jbloomAus/SAELens#202]
### Proposal 

Support the auxiliary loss function for avoiding dead neurons in top-k architectured SAEs described [here](https://cdn.openai.com/papers/sparse-autoencoders.pdf).

### Motivation

We support top-K SAEs as of #201 but haven't implemented the auxiliary loss. 

### Pitch

Implement the loss. Hopefully it's not too messy. We are probably due for some refactoring anyway but I expect this shouldn't be required to get this working. 

> We find two important ingredients for preventing dead latents: we initialize the encoder to the transpose of the
decoder, and we use an auxiliary loss that models reconstruction error using the top-kaux dead latents
(see Section A.2 for more details). Using these techniques, even in our largest (16 million latent)
autoencoder only 7% of latents are dead.

<img width="588" alt="Screenshot 2024-07-01 at 4 28 39 PM" src="https://github.com/jbloomAus/SAELens/assets/69127271/61eac0bf-ce4f-4bf1-bac9-b72dd31053b8">

### Alternatives

NA

### Additional context

- I'd expect some benchmark runs that show the training works. Can compare to stats OAI gets on GPT2 small to show it's working.
- This loss will cost us some training performance but there's no alternative. 
- Config wise I think it might make sense to move from `use_ghost_grads` being a bool to it being a Literal and having a value for this procedure (which is only valid if it's a top-K SAE). We can handle loading old configs in the pretrained SAE loading code to maintain backward compatibility with saved configs. 

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/jbloomAus/SAELens/issues) in the repo (**required**)
